File: DotNet/CesiumLanguageWriter/Advanced/ICesiumWriterAdaptor.cs
namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// Interface for classes that adapt a class derived from <see cref="CesiumPropertyWriter{TDerived}"/>.
    /// </summary>
    /// <typeparam name="TFrom">The type of property writer class to adapt.</typeparam>
    public interface ICesiumWriterAdaptor<TFrom>
    {
        /// <summary>
        /// Gets the parent being adapted.
        /// </summary>
        TFrom Parent { get; }
    }
}

File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/ICesiumWriterAdaptor.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;
import javax.annotation.Nonnull;

/**
 *  
 Interface for classes that adapt a class derived from {@link CesiumPropertyWriter}.
 
 

 * @param <TFrom> The type of property writer class to adapt.
 */
@SuppressWarnings( {
        "unused",
        "deprecation",
        "serial"
})
public interface ICesiumWriterAdaptor<TFrom> {
    /**
    *  Gets the parent being adapted.
    

    */
    @Nonnull
    TFrom getParent();
}